src/Airship/Types.hs

```diff
@@ -140,6 +141,10 @@ instance MonadBaseControl b m => MonadBaseControl b (Webmachine m) where
 -- | A convenience synonym that writes the @Monad@ type constraint for you.
 type Handler m a = Monad m => Webmachine m a
 
+-- | A constraint alias for Airship use cases that require @MonadIO@,
+-- | @MonadBaseControl@, and @MonadState@ constraints.
+type IOStateControl b s m = (MonadIO m, MonadBaseControl b m, MonadState s m)
```

Comment:
Good question. I had `b` hard coded to `IO` initially, but then I second guessed myself in favor of being more flexible. I could replace `b` with `IO` and if anyone complains about the restriction we could revisit it.